controller/config.go

```diff
@@ -1208,12 +1207,6 @@ func Validate(c Config) error {
 		}
 	}
 
-	if v, ok := c[ControllerName].(string); ok {
-		if !names.IsValidControllerName(v) {
-			return errors.Errorf("%s value must be a valid controller name (lowercase or digit with non-leading hyphen), got %q", ControllerName, v)
-		}
-	}
```

Comment:
We should still validate this, so upon bootstrap it should still be valid.

https://github.com/juju/juju/blob/1eadc941cf64c3a0652583c8e65d870e7cccf111/controller/config.go#L549-L576

Reply:
Oups, I though i had triggered bootstrap TU and didn't see any failure. I was wrong.

Reply:
I get why i do that. What bugs me was the fact that trying to setup the controller-name with an incorrect value trigger a validation error instead a read-only error (which make no sense).

I think it is due to a redundant test in https://github.com/juju/juju/blob/ec34b25e29d692c4894fdd5b005f3384cb77c682/cmd/juju/controller/config.go#L247

(NewConfig trigger a Check.Coerce on all args, even read-only even if they were checked in the loop which deal with allowed/read-only fields)

I removed the double check and it seems to works more consistently.

Reply:
Nop, unfortunately, that was not a double check, even if it looks like a double check... I added a comment to avoid such mistake in the future, and i will see if I can have a satisfying way to check readonly before validity.